cmake/target_toolchain.cmake

```diff
@@ -39,8 +39,10 @@ endif()
 # The 'generic' compiler and the 'target' compiler might be different,
 # so we unset the 'generic' one and thereby force the 'target' to
 # re-set it.
-unset(CMAKE_C_COMPILER)
-unset(CMAKE_C_COMPILER CACHE)
+if (NOT FIRST_BOILERPLATE_EXECUTION)
```

Comment:
The above comment needs to be expanded to explain the if-statement.

Reply:
Something line

```
# The 'generic' compiler and the 'target' compiler might be different,
# so we unset the 'generic' one and thereby force the 'target' to
# re-set it. This is only needed for the first boilerplate execution as
# the remaining boilerplate executions will use the same C compiler.
```
?

Reply:
(Y)

Reply:
LGTM